daliuge-engine/test/apps/test_dynlib.py

```diff
@@ -102,6 +102,7 @@ def test_cancel_dynlibprocapp(self):
         """Checks that we can cancel a long-running dynlib proc app"""
 
         a = DynlibProcApp("a", "a", lib=_libpath, sleep_seconds=10)
+        a._rpc_server = True
```

Comment:
I'm not quite sure how this test even passed before without this since DynlibProcApp cannot run if this is not set. Possibly cancel() was changing the state from ERROR to CANCELED?